### DISCUSSION TITLE
Not able align child node

### DISCUSSION THREAD

**Discussion by sathya-1994** [eclipse-glsp/glsp#455] [General] [answered]
I am using GLSP vscode extension(https://github.com/eclipse-glsp/glsp-vscode-integration) and I am generating graph using GModel.

I tried to configure gap between child and parent node with different options(GLayoutOptions.KEY_PADDING_BOTTOM,GLayoutOptions.KEY_PADDING_TOP,GLayoutOptions.KEY_PADDING_LEFT,GLayoutOptions.KEY_PADDING_RIGHT,GLayoutOptions.KEY_H_GA,GLayoutOptions.KEY_V_GAP). But I am not able to place child node at top.

**Screenshot:**
![image](https://user-images.githubusercontent.com/59145404/141610892-a1f53a9d-b101-4efc-8f57-3e82191df2a9.png)

**Expected Layout:**
![image](https://user-images.githubusercontent.com/59145404/141610768-261da210-5515-42b5-8d67-35f5e07f7624.png)

I tired with combination of GNodeImpl and GCompartmentImpl. But I was not able to create above layout.

**Where I can find documentation related to node alignments?**

**Previously I worked with Eclipse ELK, there we can configure node-node, node-edge, parent node - child node gaps/separations. How we can configure these values in GLSP?**

**Graph Data:**
GGraph graph = new GGraphImpl();
graph.setId(DefaultTypes.GRAPH + Integer.toString(graphCount++));
graph.setType(DefaultTypes.GRAPH);
GNodeImpl gNode = new GNodeImpl();
gNode.setParent(graph);
GDimensionImpl d = new GDimensionImpl();
d.setHeight(50);
d.setWidth(50);
gNode.setType("node");
gNode.setSize(d);
gNode.setId("node1");
			
GNodeImpl gNode2 = new GNodeImpl();
gNode2.setParent(gNode);
GDimensionImpl d2 = new GDimensionImpl();
d2.setHeight(100);
d2.setWidth(100);
gNode2.setType("node");
			
// Added new css class newNode
gNode2.getCssClasses().add("**newNode**");
gNode2.setSize(d2);
gNode2.setId("node3");

**We have to add below css class to GLSP vscode extension.
CSS class:
.newNode>.sprotty-node {
          fill: green;
}**

**Comment by CamilleLetavernier** [comment, marked as answer]
Hi,

Did you try setting the parent node layout to VBox?

There has been some recent changes to the VBox layout; if you're using a recent nightly build, I think this case is supported. There are still a few limitations with the current layouts, but additional options may be added. In this case, I believe combining a VBox layout with some Top-Lef-Right Paddings should do the trick.
